engine/gamesys/src/gamesys/components/comp_collision_object.cpp

```diff
@@ -2099,6 +2097,24 @@ namespace dmGameSystem
         return true;
     }
 
+    void UpdateMass(void* _world, void* _component, float mass)
```

Comment:
I would still like it to be called "SetMass" like any other setter/getter.

Reply:
I explained it in the DD:
>Using of `set_mass` feels like wrong expectations
>Current function is more like `update_mass`
because it’s not replacing one value with another that will be taken by a system later
>But this is an active action where we iterate over all collision body fixtures calculating density and after that reseting mass of the collision body to recalculate the rest of parameters based on fixtures with new density

If I change the other APIs we have then with `set_*` we just change value in a structure but in this case we start recalculation process

Reply:
there will de another function later, it will be something like `set_mass_data` and `get_mass_data` - these two are getter/seeter functions. Because they just set and force using of values from it without any calculations and ignoring physic laws

Reply:
more info about it here: https://docs.google.com/document/d/1BJDYFRNSAiRdKhUQeaH6MkwH0ok_XW851R6-Rne7zQ8/edit

Reply:
Sorry I didn't see that part of the DD. And I don't wholly agree with that :)
I can agree that it might be a way to communicate that some extra work is being done.
But I also think that "Set" functions are allowed to calculate things, just as "Get" functions are allowed to calcualte things.
But, let's go with this.